sklearn_numba_dpex/common/kernels.py

```diff
@@ -429,17 +395,26 @@ def fused_elementwise_func_(x):
     if fused_elementwise_func is None:
         partial_sum_reduction_nofunc = partial_sum_reduction
     else:
-        _, partial_sum_reduction_nofunc = _make_partial_sum_reduction_2d_axis1_kernel(
+        *_, partial_sum_reduction_nofunc = _make_partial_sum_reduction_2d_axis1_kernel(
             n_rows, work_group_size, fused_elementwise_func_, dtype
         )
 
-    return work_group_size, (
-        (partial_sum_reduction, partial_sum_reduction_nofunc),  # kernels
-        reduction_block_size,
+    get_result_shape = lambda result_sum_axis_size: (n_rows, result_sum_axis_size)
+    get_global_size = lambda result_sum_axis_size: (
+        n_rows,
+        result_sum_axis_size * work_group_size,
     )
 
+    return (
+        work_group_shape,
+        (
+            (partial_sum_reduction, partial_sum_reduction_nofunc),  # kernels
+            reduction_block_size,
+        ),
+        (get_result_shape, get_global_size),
+    )  # shape_update_fn
```

Comment:
The last comment likely is off due to black formatting.

What do you think of this suggestion?
```suggestion
    kernels = (partial_sum_reduction, partial_sum_reduction_nofunc)
    shape_update_fn = (get_result_shape, get_global_size)
    return (
        work_group_shape,
        (
            kernels,
            reduction_block_size,
        ),
        shape_update_fn,
    )
```